lib/core/network.dart

```diff
@@ -22,21 +35,45 @@ class Network {
         },
       ),
     );
-    _dio.interceptors.add(RequestIdInterceptor());
+    _enteDio = Dio(
+      BaseOptions(
+        baseUrl: apiEndpoint,
+        connectTimeout: kConnectTimeout,
+        headers: {
+          HttpHeaders.userAgentHeader: FkUserAgent.userAgent,
+          'X-Client-Version': packageInfo.version,
+          'X-Client-Package': packageInfo.packageName,
+        },
+      ),
+    );
+    enteDio.interceptors.add(EnteRequestInterceptor(preferences));
   }
 
   Network._privateConstructor();
 
   static Network instance = Network._privateConstructor();
 
   Dio getDio() => _dio;
+
+  Dio get enteDio => _enteDio;
 }
 
-class RequestIdInterceptor extends Interceptor {
+class EnteRequestInterceptor extends Interceptor {
+  final SharedPreferences preferences;
```

Comment:
`_preferences` since it's private?

P.S: I also find this forced `_` prefix pattern irritating :(